tests/robustness/snapmeta/kopia_meta.go

```diff
@@ -0,0 +1,127 @@
+package snapmeta
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	"github.com/kopia/kopia/tests/tools/kopiarunner"
+)
+
+var _ Store = &KopiaMetadata{}
+var _ DataPersister = &KopiaMetadata{}
+
+// KopiaMetadata handles metadata persistency of a snapshot store, using a Kopia
+// repository as the persistency mechanism
+type KopiaMetadata struct {
+	*Simple
+	LocalMetadataDir string
+	snap             *kopiarunner.KopiaSnapshotter
+}
+
+// NewKopiaMetadata instantiates a new KopiaMetadata and returns its pointer.
+func NewKopiaMetadata() (*KopiaMetadata, error) {
```

Comment:
Since we are going with the interface, then it makes sense to return an interface here instead of the specific type. Then the type would not need to be exported, it is a particular implementation.

```suggestion
func New() (Store, error) {
```

Also, see the related questions below about the interfaces, because for the suggestion above to work, this would have to be a single interface.